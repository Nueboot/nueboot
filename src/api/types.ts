export type Material = 'Knit' | 'Leather' | 'Synthetic';
export type Brands =
  | 'Adidas'
  | 'Nike'
  | 'Puma'
  | 'New Balance'
  | 'Under Armour'
  | 'Mizuno';

export interface BootInfo {
  brand: Brands;
  material: Material;
  midcut: boolean;
  model: string;
  msrp: number;
  releaseDate: string;
  weight: number;
  id: string;
}

export interface BootReview {
  body: string;
  id: string;
  stars: number;
  user: string;
}
