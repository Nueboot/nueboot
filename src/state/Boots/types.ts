import { BootInfo } from 'api/types';

export interface BootsState {
  readonly boots: BootInfo[];
  readonly loading: boolean;
}
