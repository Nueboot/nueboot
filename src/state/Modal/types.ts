export interface ModalState {
  modalType: ModalTypes | null;
}

export type ModalTypes = 'login' | 'signup' | 'reset';
