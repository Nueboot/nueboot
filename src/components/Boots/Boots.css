.boots {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  column-gap: 1rem;
}

@media (max-width: 40rem) { 
  .boots {
    grid-template-columns: repeat(2, 1fr);
  }
}
