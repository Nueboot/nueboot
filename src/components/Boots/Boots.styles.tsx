import styled from "styled-components";

export const Container = styled.div`
  margin: 40px 0 100px 0;

  h1 {
    font-size: 24px;
  }
  @media (max-width: 960px) {
    h1 {
      font-weight: 500;
      font-size: 18px;
      line-height: 26px;
    }
  }
`;

export const CardContainer = styled.div`
  display: grid;
  grid-auto-flow: row;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 14px;
  width: 100%;

  @media (max-width: 960px) {
    grid-template-columns: repeat(1, 1fr);
  }
`;

export const Heading = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  h4 {
    margin: 0;
  }
`;

export const FilterButton = styled.div`
  cursor: pointer;
`;
