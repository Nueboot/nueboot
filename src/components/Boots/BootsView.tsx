import React from 'react';

import { BootInfo } from 'api/types';
import BootCard from 'components/BootCard';
import styled from 'styled-components';

export interface BootsViewProps {
  boots: BootInfo[];
}

class BootsView extends React.Component<BootsViewProps> {
  public render() {
    const boots = this.props.boots;
    return (
      <div className="boots">
        {boots.map(boot => (
          <BootCardContainer key={boot.id}>
            <BootCard boot={boot} />
          </BootCardContainer>
        ))}
      </div>
    );
  }
}

export default BootsView;

const BootCardContainer = styled.div`
  margin-bottom: 1rem;
  padding: 0.25rem;
  :hover {
    opacity: 0.7;
  }
`;
