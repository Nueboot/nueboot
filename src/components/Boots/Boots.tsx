import React from 'react';

import { BootInfo } from '../../api/types';
import { Container, Heading, NavigationLink } from '../Styled';
import Spinner from '../Styled/Spinner';

import './Boots.css';

export interface StateProps {
  boots: BootInfo[];
  loading: boolean;
}

export type BootsProps = StateProps;

export default class Boots extends React.Component<BootsProps> {
  public render() {
    if (this.props.loading) {
      return <Spinner />;
    }
    if (!this.props.boots) {
      return null;
    }
    return (
      <Container>
        <Heading>Boots</Heading>
        <div className="boots">
          {Object.keys(this.props.boots).map(key => {
            const boot: BootInfo = this.props.boots[key];
            return (
              <div className="boot-list-item" key={boot.model}>
                <NavigationLink to={`boots/${key}`}>
                  {boot.brand} {boot.model}
                </NavigationLink>
              </div>
            );
          })}
        </div>
      </Container>
    );
  }
}
