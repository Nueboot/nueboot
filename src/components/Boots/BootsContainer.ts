import { connect } from 'react-redux';
import { allBoots, bootLoading, getBootsByBrand } from 'state/Boots/selectors';
import { State } from 'state/types';
import Boots, { StateProps } from './Boots';

export const mapStateToProps = (state: State): StateProps => ({
  boots: allBoots(state),
  byBrand: getBootsByBrand(state),
  loading: bootLoading(state),
});

export default connect(mapStateToProps)(Boots);
