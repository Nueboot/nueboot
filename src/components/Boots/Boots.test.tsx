import { RouteComponentProps } from '@reach/router';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { Boot } from '../../types';
import Boots from './Boots';

jest.mock('../../api', () => ({
  getAllBoots: () => Promise.resolve(null),
}));

describe('Boots', () => {
  let component: ShallowWrapper<RouteComponentProps>;

  describe('with no boots', () => {
    beforeEach(() => {
      component = shallow(<Boots />);
    });

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with boots', () => {
    let boot: Boot;
    beforeEach(() => {
      boot = {
        brand: 'Adidas',
        material: 'Knit',
        midcut: true,
        model: 'Predator',
        msrp: 100,
        releaseDate: '1/1/2001',
        weight: 99,
      };
      component.setState({
        boots: [
          boot,
        ],
      });
    });

    it('renders correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
