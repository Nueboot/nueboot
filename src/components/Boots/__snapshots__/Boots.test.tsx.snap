// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boots with boots renders correctly 1`] = `
<styled.div>
  <styled.h1>
    Boots
  </styled.h1>
  <div
    key="Predator"
  >
    <ForwardRef
      state={
        Object {
          "boot": Object {
            "brand": "Adidas",
            "material": "Knit",
            "midcut": true,
            "model": "Predator",
            "msrp": 100,
            "releaseDate": "1/1/2001",
            "weight": 99,
          },
        }
      }
      to="/boots/1"
    >
      <styled.span
        bold={true}
      >
        Adidas
      </styled.span>
       
      <styled.span>
        Predator
      </styled.span>
    </ForwardRef>
  </div>
</styled.div>
`;

exports[`Boots with no boots renders correctly 1`] = `<Spinner />`;
